src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,99 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './TodoList.scss';
+import { Todo } from '../../react-app-env';
+import { getAllTodos } from '../../api';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+interface Props {
+  selectedUserId: number,
+  chahgeSelectedUserId: (id: number) => void,
+}
+
+enum Options {
+  all,
+  active,
+  completed,
+}
+
+export const TodoList: React.FC<Props> = (props) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [input, setInput] = useState('');
+  const [status, setStatus] = useState(Options.all);
+
+  useEffect(() => {
+    getAllTodos()
+      .then(response => setTodos(response));
+  }, []);
+
+  const handleOptions = (filterOption: Options, todo: Todo) => {
+    switch (filterOption) {
+      case Options.active:
+        return todo.completed === false;
+      case Options.completed:
+        return todo.completed === true;
+      default:
+        return true;
+    }
+  };
+
+  const filteredTodos = todos
+    .filter(todo => todo.title.toLowerCase().includes(input.toLowerCase())
+    && handleOptions(status, todo));
+
+  return (
+    <div className="TodoList">
+      <h2>Todos:</h2>
+
+      <div className="TodoList__list-container">
+
+        <input
+          type="text"
+          value={input}
+          onChange={(event) => {
+            setInput(event.target.value);
+          }}
+        />
+
+        <select
+          value={status}
+          onChange={(event) => {
+            setStatus(+event.target.value);
```

Comment:
remove `+` use `as` instead
```suggestion
            setStatus(event.target.value as Options);
```

Reply:
setStatus(event.target.value as unknown as Options) - It works only this way

Reply:
"setStatus(event.target.value as Options)" and "setStatus(event.target.value as unknown as Options)" dont work

 
error for "setStatus(event.target.value as Options)"  - Преобразование типа "string" в тип "Options" может привести к ошибке, так как ни один из типов не перекрывается с другим в достаточной степени. Если это сделано намеренно, сначала преобразуйте выражение в "unknown".ts(2352)

error for "setStatus(event.target.value as unknown as Options)" - selecting an option does not affect the list

Reply:
create enum like this and `as` will work
```
enum Options {
  all: 'all',
  active: 'active',
  completed: 'completed',
}
```